clients/lsp-svelte.el

```diff
@@ -37,17 +37,263 @@
+  "Which css files should be checked for global variables (`--global-var: value;`). These variables are added to the css completions. String of comma-separated file paths or globs relative to workspace root."
+  :type 'string
+  :package-version '(lsp-mode . "7.1.0"))
+
+(defcustom lsp-svelte-plugin-css-diagnostics-enable t
+  "Enable diagnostic messages for CSS"
+  :type 'boolean
+  :package-version '(lsp-mode . "7.1.0"))
+
+(defcustom lsp-svelte-plugin-css-hover-enable t
+  "Enable hover info for CSS"
+  :type 'boolean
+  :package-version '(lsp-mode . "7.1.0"))
+
+(defcustom lsp-svelte-plugin-css-completions-enable t
+  "Enable auto completions for CSS"
+  :type 'boolean
+  :package-version '(lsp-mode . "7.1.0"))
+
+(defcustom lsp-svelte-plugin-css-completions-emmet t
+  "Enable emmet auto completions for CSS"
+  :type 'boolean
+  :package-version '(lsp-mode . "7.1.0"))
+
+(defcustom lsp-svelte-plugin-css-document-colors-enable t
+  "Enable document colors for CSS"
+  :type 'boolean
+  :package-version '(lsp-mode . "7.1.0"))
+
+(defcustom lsp-svelte-plugin-css-color-presentations-enable t
+  "Enable color picker for CSS"
+  :type 'boolean
+  :package-version '(lsp-mode . "7.1.0"))
+
+(defcustom lsp-svelte-plugin-css-document-symbols-enable t
+  "Enable document symbols for CSS"
+  :type 'boolean
+  :package-version '(lsp-mode . "7.1.0"))
+
+(defcustom lsp-svelte-plugin-css-selection-range-enable t
+  "Enable selection range for CSS"
+  :type 'boolean
+  :package-version '(lsp-mode . "7.1.0"))
+
+(defcustom lsp-svelte-plugin-html-enable t
+  "Enable the HTML plugin"
+  :type 'boolean
+  :package-version '(lsp-mode . "7.1.0"))
+
+(defcustom lsp-svelte-plugin-html-hover-enable t
+  "Enable hover info for HTML"
+  :type 'boolean
+  :package-version '(lsp-mode . "7.1.0"))
+
+(defcustom lsp-svelte-plugin-html-completions-enable t
+  "Enable auto completions for HTML"
+  :type 'boolean
+  :package-version '(lsp-mode . "7.1.0"))
+
+(defcustom lsp-svelte-plugin-html-completions-emmet t
+  "Enable emmet auto completions for HTML"
+  :type 'boolean
+  :package-version '(lsp-mode . "7.1.0"))
+
+(defcustom lsp-svelte-plugin-html-tag-complete-enable t
+  "Enable HTML tag auto closing"
+  :type 'boolean
+  :package-version '(lsp-mode . "7.1.0"))
+
+(defcustom lsp-svelte-plugin-html-document-symbols-enable t
+  "Enable document symbols for HTML"
+  :type 'boolean
+  :package-version '(lsp-mode . "7.1.0"))
+
+(defcustom lsp-svelte-plugin-svelte-enable t
+  "Enable the Svelte plugin"
+  :type 'boolean
+  :package-version '(lsp-mode . "7.1.0"))
+
+(defcustom lsp-svelte-plugin-svelte-diagnostics-enable t
+  "Enable diagnostic messages for Svelte"
+  :type 'boolean
+  :package-version '(lsp-mode . "7.1.0"))
+
+(defcustom lsp-svelte-plugin-svelte-compiler-warnings nil
+  "Svelte compiler warning codes to ignore or to treat as errors.
+Example: { 'css-unused-selector': 'ignore', 'unused-export-let':
+'error'}"
+  :type 'plist
```

Comment:
Confirmed working with this amendment to the compiler warning format! Thanks so much!

```suggestion
  :type '(alist :key-type (string :tag "Warning code")
                :value-type (choice
                             (const :tag "Ignore" "ignore")
                             (const :tag "Treat as error" "error")))
```

Reply:
Applied.

Reply:
Actually, native JSON cannot serialize alists with string keys. We should either change keys to symbols or use hts.